benchmarks/asv.conf.json

```diff
@@ -44,7 +44,7 @@
     // If missing or the empty string, the tool will be automatically
     // determined by looking for tools on the PATH environment
     // variable.
-    "environment_type": "mamba",
+    "environment_type": "conda",
```

Comment:
?

Reply:
no reason to use mamba since conda now uses the mamba solver by default

Reply:
I had to upgrade my conda so that it would use the lib mamba solver by default, and now I get an error here:

```sh
$ asv run HEAD --steps 1
Couldn't load asv.plugins._mamba_helpers because
No module named 'libmambapy'
· Creating environments
·· Error running /mnt/workspace/mambaforge/bin/conda env create -f /mnt/workspace/tmp/tmpd4g2uh44.yml -p /mnt/workspace/repos/anndata/benchmarks/.asv/env/64c601b255f4a31e6994caef8c345b33 --force (exit status 2)
   STDOUT -------->
   
   STDERR -------->
   usage: conda [-h] [-v] [--no-plugins] [-V] COMMAND ...
   conda: error: unrecognized arguments: --force

·· Failure creating environment for conda-py3.12-h5py-memory_profiler-natsort-numpy-pandas-pooch-pytest-pytoml-scipy-zarr
Traceback (most recent call last):
  File "/mnt/workspace/mambaforge/envs/anndata-dev/bin/asv", line 8, in <module>
    sys.exit(main())
             ^^^^^^
  File "/mnt/workspace/mambaforge/envs/anndata-dev/lib/python3.12/site-packages/asv/main.py", line 29, in main
    result = args.func(args)
             ^^^^^^^^^^^^^^^
  File "/mnt/workspace/mambaforge/envs/anndata-dev/lib/python3.12/site-packages/asv/commands/__init__.py", line 49, in run_from_args
    return cls.run_from_conf_args(conf, args)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/mnt/workspace/mambaforge/envs/anndata-dev/lib/python3.12/site-packages/asv/commands/run.py", line 166, in run_from_conf_args
    return cls.run(
           ^^^^^^^^
  File "/mnt/workspace/mambaforge/envs/anndata-dev/lib/python3.12/site-packages/asv/commands/run.py", line 278, in run
    Setup.perform_setup(environments, parallel=parallel)
  File "/mnt/workspace/mambaforge/envs/anndata-dev/lib/python3.12/site-packages/asv/commands/setup.py", line 82, in perform_setup
    list(map(_create, environments))
  File "/mnt/workspace/mambaforge/envs/anndata-dev/lib/python3.12/site-packages/asv/commands/setup.py", line 14, in _create
    env.create()
  File "/mnt/workspace/mambaforge/envs/anndata-dev/lib/python3.12/site-packages/asv/environment.py", line 750, in create
    self._setup()
  File "/mnt/workspace/mambaforge/envs/anndata-dev/lib/python3.12/site-packages/asv/plugins/conda.py", line 149, in _setup
    self._run_conda(['env', 'create', '-f', env_file_name,
  File "/mnt/workspace/mambaforge/envs/anndata-dev/lib/python3.12/site-packages/asv/plugins/conda.py", line 203, in _run_conda
    return util.check_output([conda] + args,
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/mnt/workspace/mambaforge/envs/anndata-dev/lib/python3.12/site-packages/asv/util.py", line 618, in check_output
    raise ProcessError(args, retcode, stdout, stderr)
asv.util.ProcessError: Command '/mnt/workspace/mambaforge/bin/conda env create -f /mnt/workspace/tmp/tmpd4g2uh44.yml -p /mnt/workspace/repos/anndata/benchmarks/.asv/env/64c601b255f4a31e6994caef8c345b33 --force' returned non-zero exit status 2
```

I think this comes from the change:

> * Remove conda env create --force. Use conda env create --yes instead. (#12681 via #13634)

from 24.3.0 (https://conda.io/projects/conda/en/latest/release-notes.html#id1)


We aren't seeing this on the CI job because it's running an old version of conda, which also happens to be too old to use the mamba solver (the change in default came out in [23.10](https://conda.io/projects/conda/en/latest/release-notes.html#id23))

<details>
<summary> warning from CI job </summary>

```
   ==> WARNING: A newer version of conda exists. <==
     current version: 23.9.0
     latest version: 24.3.0
   
   Please update conda by running
   
       $ conda update -n base -c conda-forge conda
   
   Or to minimize the number of packages updated during conda update use
   
        conda install conda=24.3.0
```   
   

</details>


I'd suggest we just stick with mamba for a bit. Either asv needs to update or conda needs to roll back

Reply:
Also if I change this back to mamba locally it runs fine

Reply:
```suggestion
    "environment_type": "mamba",
```